pkg/apiserver/resthandler.go

```diff
@@ -171,6 +172,85 @@ func CreateResource(r RESTCreater, ctxFn ContextFunc, namer ScopeNamer, codec ru
 	}
 }
 
+// PatchResource returns a function that will handle a resource patch
+func PatchResource(r RESTPatcher, ctxFn ContextFunc, namer ScopeNamer, codec runtime.Codec, resource string, admit admission.Interface) restful.RouteFunction {
+	return func(req *restful.Request, res *restful.Response) {
+		w := res.ResponseWriter
+
+		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
+		timeout := parseTimeout(req.Request.URL.Query().Get("timeout"))
+
+		namespace, name, err := namer.Name(req)
+		if err != nil {
+			notFound(w, req.Request)
+			return
+		}
+		ctx := ctxFn(req)
+		ctx = api.WithNamespace(ctx, namespace)
+
+		original, err := r.Get(ctx, name)
```

Comment:
Yeah - same thought here.  We needed that on client update as well (the way merge is handled is ugly right now).


> On Feb 23, 2015, at 4:30 PM, Daniel Smith <notifications@github.com> wrote:
> 
> In pkg/apiserver/resthandler.go:
> 
> > +func PatchResource(r RESTPatcher, ctxFn ContextFunc, namer ScopeNamer, codec runtime.Codec, resource string, admit admission.Interface) restful.RouteFunction {
> > +	return func(req *restful.Request, res *restful.Response) {
> > +		w := res.ResponseWriter
> > +
> > +		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
> > +		timeout := parseTimeout(req.Request.URL.Query().Get("timeout"))
> > +
> > +		namespace, name, err := namer.Name(req)
> > +		if err != nil {
> > +			notFound(w, req.Request)
> > +			return
> > +		}
> > +		ctx := ctxFn(req)
> > +		ctx = api.WithNamespace(ctx, namespace)
> > +
> > +		original, err := r.Get(ctx, name)
> Agree with @smarterclayton. There's probably some refactoring of Update in which we pass in a function that takes the old object with status removed etc and merges the new object, such that both PATCH and PUT could produce such a function.
> 
> —
> Reply to this email directly or view it on GitHub.
>